airflow/www/utils.py

```diff
@@ -779,6 +779,80 @@ class UtcAwareFilterConverter(fab_sqlafilters.SQLAFilterConverter):
     """Retrieve conversion tables for UTC-Aware filters."""
 
 
+class XComFilterStartsWith(fab_sqlafilters.FilterStartsWith):
+    """Starts With filter for XCom values."""
+
+    def apply(self, query, value):
+        query, field = get_field_setup_query(query, self.model, self.column_name)
```

Comment:
The code here looks similar to others in the `apply` method. We probably could add a function to reduce code duplication. Also it would be nice if we could add type annotation. Thanks!

Reply:
I centralized the code for handling value filtering in the `apply()` method into the `get_trimmed_field()` function to reduce code duplication. Additionally, I added type annotations for `apply()` and `get_trimmed_field()`